modules/client/screen/src/main/java/net/legacyfabric/fabric/api/client/screen/v1/ScreenMouseEvents.java

```diff
@@ -0,0 +1,273 @@
+	}
+
+	/**
+	 * An event that is called after the release of a mouse click is processed for a screen.
+	 *
+	 * @return the event
+	 */
+	public static Event<AfterMouseRelease> afterMouseRelease(Screen screen) {
+		Objects.requireNonNull(screen, "Screen cannot be null");
+
+		return ScreenExtensions.getExtensions(screen).fabric_getAfterMouseReleaseEvent();
+	}
+
+	/**
+	 * An event that is checks if the mouse should be allowed to scroll in a screen.
+	 *
+	 * <p>This event tracks amount of vertical and horizontal scroll.
+	 *
+	 * @return the event
+	 */
+	public static Event<AllowMouseScroll> allowMouseScroll(Screen screen) {
+		Objects.requireNonNull(screen, "Screen cannot be null");
+
+		return ScreenExtensions.getExtensions(screen).fabric_getAllowMouseScrollEvent();
+	}
+
+	/**
+	 * An event that is called after mouse scrolling is processed for a screen.
+	 *
+	 * <p>This event tracks amount of vertical and horizontal scroll.
+	 *
+	 * @return the event
+	 */
+	public static Event<BeforeMouseScroll> beforeMouseScroll(Screen screen) {
+		Objects.requireNonNull(screen, "Screen cannot be null");
+
+		return ScreenExtensions.getExtensions(screen).fabric_getBeforeMouseScrollEvent();
+	}
+
+	/**
+	 * An event that is called after mouse scrolling is processed for a screen.
+	 *
+	 * <p>This event tracks amount a mouse was scrolled both vertically and horizontally.
+	 *
+	 * @return the event
+	 */
+	public static Event<AfterMouseScroll> afterMouseScroll(Screen screen) {
+		Objects.requireNonNull(screen, "Screen cannot be null");
+
+		return ScreenExtensions.getExtensions(screen).fabric_getAfterMouseScrollEvent();
+	}
+
+	private ScreenMouseEvents() {
+	}
+
+	@Environment(EnvType.CLIENT)
+	@FunctionalInterface
+	public interface AllowMouseClick {
+		/**
+		 * @param mouseX the x position of the mouse
+		 * @param mouseY the y position of the mouse
+		 * @param button the button number, which can be identified by the constants in {@link org.lwjgl.input.Mouse Mouse}.
+		 * @see org.lwjgl.input.Mouse
+		 */
+		boolean allowMouseClick(Screen screen, double mouseX, double mouseY, int button);
+	}
+
+	@Environment(EnvType.CLIENT)
+	@FunctionalInterface
+	public interface BeforeMouseClick {
+		/**
+		 * @param mouseX the x position of the mouse
+		 * @param mouseY the y position of the mouse
+		 * @param button the button number, which can be identified by the constants in {@link org.lwjgl.input.Mouse Mouse}.
+		 * @see org.lwjgl.input.Mouse
+		 */
+		void beforeMouseClick(Screen screen, double mouseX, double mouseY, int button);
+	}
+
+	@FunctionalInterface
+	@Environment(EnvType.CLIENT)
+	public interface AfterMouseClick {
+		/**
+		 * @param mouseX the x position of the mouse
+		 * @param mouseY the y position of the mouse
+		 * @param button the button number, which can be identified by the constants in {@link org.lwjgl.glfw.GLFW GLFW}.
+		 * @see org.lwjgl.glfw.GLFW#GLFW_MOUSE_BUTTON_1
```

Comment:
might want to remove/change these

Reply:
All of these JavaDoc comments need proper fixes that I'm not ready for. I haven't implemented the mixins for most of the input events because of the lack of GLFW.

Reply:
@DeadlyMC should have a look at this now. I've just pushed what I believe to be a valid equivalent.